chaos_kitten/brain/orchestrator.py

```
@@ -122,21 +127,15 @@ async def execute_and_analyze(
+                llm = ChatAnthropic(model=model, temperature=temperature)
             else:
                 raise ValueError(f"Unsupported LLM provider for adaptive mode: {provider}")
                 
-            if adaptive_mode and llm:
-                adaptive_gen = AdaptivePayloadGenerator(llm, max_rounds=max_rounds)
-        except ImportError as e:
-            logger.error(f"Failed to import LLM provider dependencies: {e}")
-            logger.warning("Adaptive mode disabled due to missing dependencies.")
+            adaptive_gen = AdaptivePayloadGenerator(llm, max_rounds=max_rounds)
+        except (ImportError, ValueError) as e:
+            logger.exception("Failed to set up adaptive LLM: %s", e)
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)